src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyValidators/OrganizationDataOwnershipPolicyValidator.cs

```diff
@@ -26,6 +26,33 @@ public class OrganizationDataOwnershipPolicyValidator(
 
     public override Task<string> ValidateAsync(PolicyUpdate policyUpdate, Policy? currentPolicy) => Task.FromResult("");
 
+    public override Task ProtoTypeOnSaveSideEffectsAsync(SavePolicyRequest policyRequest, Policy? currentPolicy)
+    {
+        var metadata = MapToOrganizationModelOwnershipPolicyModel(policyRequest.Metadata);
+
+
+        // Here we will have command data associated with this particular policy validator.
+        if (metadata!.ExecuteSideEffect)
+        {
+
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public static OrganizationModelOwnershipPolicyModel? MapToOrganizationModelOwnershipPolicyModel(IPolicyMetadataModel model)
```

Comment:
@eliykat Here we need a mapper to turn the generic metadata object into something the validator can use. We can easily centralize this in a switch statement or something similar.

Reply:
I'd like to see how this is actually handled in the flow from client -> controller -> command -> validator.

Given that this model depends on the policy type, the controller won't know how to instantiate it. One example of how to handle this is in the `Policy.data` property: it's a json blob, it's maintained as a `string` through the api layer, and then the policy specific code uses `CoreHelpers.LoadClassFromJsonData` to instantiate it into an object when needed.

You could follow a similar pattern here, maintaining it as a string through to the validator, then the validator creates the actual object to deal with it. However, that does tightly couple it to the request sent by the client, whereas the `executeSideEffect` property is set by the controller layer in your current code.